coldfront/plugins/qumulo/utils/eib_billing.py

```diff
@@ -0,0 +1,187 @@
+        WHEN 'condo' THEN CEILING(CAST(storage_quota AS FLOAT8) /500)
+      END billing_unit,
+      CASE service_rate_category
+        WHEN 'consumption' THEN 'TB'
+        WHEN 'subscription' THEN '100TB'
+        WHEN 'subscription_500tb' THEN '500TB'
+        WHEN 'condo' THEN '500TB'
+      END service_unit,
+      storage_filesystem_path
+    FROM allocation_allocation a
+    JOIN allocation_allocationstatuschoice astatus ON a.status_id=astatus.id
+    JOIN project_project p ON a.project_id=p.id
+    JOIN auth_user u ON p.pi_id=u.id
+    LEFT JOIN (SELECT aa.allocation_id, aa.value storage_name FROM allocation_allocationattribute aa JOIN allocation_allocationattributetype aat ON aa.allocation_attribute_type_id=aat.id WHERE aat.name='storage_name') AS storage_name ON a.id=storage_name.allocation_id
+    LEFT JOIN (SELECT aa.allocation_id, aa.value storage_filesystem_path FROM allocation_allocationattribute aa JOIN allocation_allocationattributetype aat ON aa.allocation_attribute_type_id=aat.id WHERE aat.name='storage_filesystem_path') AS storage_filesystem_path ON a.id=storage_filesystem_path.allocation_id
+    LEFT JOIN (SELECT aa.allocation_id, aa.value cost_center FROM allocation_allocationattribute aa JOIN allocation_allocationattributetype aat ON aa.allocation_attribute_type_id=aat.id WHERE aat.name='cost_center') AS cost_center ON a.id=cost_center.allocation_id
+    LEFT JOIN (SELECT aa.allocation_id, aa.value department_number FROM allocation_allocationattribute aa JOIN allocation_allocationattributetype aat ON aa.allocation_attribute_type_id=aat.id WHERE aat.name='department_number') AS department_number ON a.id=department_number.allocation_id
+    LEFT JOIN (SELECT aa.allocation_id, aa.value service_rate_category FROM allocation_allocationattribute aa JOIN allocation_allocationattributetype aat ON aa.allocation_attribute_type_id=aat.id WHERE aat.name='service_rate') AS service_rate_category ON a.id=service_rate_category.allocation_id
+    LEFT JOIN (SELECT aa.allocation_id, aa.id, aa.value storage_quota FROM allocation_allocationattribute aa JOIN allocation_allocationattributetype aat ON aa.allocation_attribute_type_id=aat.id WHERE aat.name='storage_quota') AS storage_quota ON a.id=storage_quota.allocation_id
+    JOIN (
+      SELECT allocation_attribute_id,
+        MAX(modified) recent_modified,
+        value
+      FROM allocation_historicalallocationattributeusage
+      GROUP BY allocation_attribute_id, DATE(modified)
+    ) AS most_recent_haau
+      ON storage_quota.id=most_recent_haau.allocation_attribute_id
+    JOIN allocation_allocation_resources ar ON ar.allocation_id=a.id
+    JOIN resource_resource r ON r.id=ar.resource_id
+    WHERE
+      r.name = 'Storage2'
+    AND
+      DATE(most_recent_haau.recent_modified) = '%s'
+    AND
+      astatus.name = 'Active'
+  ) AS data
+) AS report where report.billing_unit > 0;
+"""
+
+filename = ""
+
+def reset_filename() -> None:
+    global filename
```

Comment:
As Jim mentioned on the review last week, this would be better refactored to use a `class` and `class` members rather than a `global` variable for filename.

If this pattern is followed by other billing utils defined in this module, it's possible we could have collisions.  Refactoring to a `class` would make that impossible as well as make the code more readable.

Reply:
Let me push with the fixes of the formats and some minor changes first. I will try to refactor it to a class in another push.

Reply:
refactored to use `class` in the latest commit.